VideoLocker/src/main/java/org/edx/mobile/module/notification/EdxParsePushBroadcastReceiver.java

```diff
@@ -1,132 +1,126 @@
-                PendingIntent resultPendingIntent =  PendingIntent.getActivity(
-                        context,
-                        0,
-                        resultIntent,
-                        PendingIntent.FLAG_UPDATE_CURRENT
-                );
-
-                NotificationManager notificationManager =
-                        (NotificationManager) context.getSystemService(Context.NOTIFICATION_SERVICE);
-
-                NotificationCompat.Builder builder = new NotificationCompat.Builder(context);
-                builder.setContentTitle(titleTemplate);
-                builder.setContentText(payload.getCourseName());
-                builder.setSmallIcon(R.drawable.app_icon);
-                builder.setContentIntent(resultPendingIntent);
-                builder.setAutoCancel(true);
-
-                Uri uri= RingtoneManager.getDefaultUri(RingtoneManager.TYPE_NOTIFICATION);
-                builder.setSound(uri);
-
-                Notification notification = builder.build();
-                notification.flags |= Notification.FLAG_AUTO_CANCEL;
-                return notification;
-            } else {
-                EventBus.getDefault().postSticky(new CourseAnnouncementEvent(
-                    CourseAnnouncementEvent.EventType.MESSAGE_RECEIVED, ""));
-                return super.getNotification(context,intent);
+    @Override
+    protected android.app.Notification getNotification(@NonNull Context context, @NonNull Intent intent) {
+        final BaseNotificationPayload payload = ParseHandleHelper.extractPayload(intent);
+        if (null != payload && null != payload.getAction()) {
+            switch (payload.getAction()) {
+                case ParseHandleHelper.COURSE_ANNOUNCEMENT_ACTION: {
+                    final CourseUpdateNotificationPayload courseUpdateNotificationPayload = (CourseUpdateNotificationPayload) payload;
+                    getSegment(context).trackNotificationReceived(courseUpdateNotificationPayload.getCourseId());
+                    return newNotificationBuilder(context, intent)
+                            .setContentTitle(context.getString(R.string.COURSE_ANNOUNCEMENT_NOTIFICATION_TITLE))
+                            .setContentText(courseUpdateNotificationPayload.getCourseName())
+                            .build();
+                }
             }
+        }
+
+        // Default behaviour for unknown notification types
+        getSegment(context).trackNotificationReceived("");
+        return super.getNotification(context, intent);
+    }
 
-        } catch (Exception e) {
-            return super.getNotification(context, intent);
+    private NotificationCompat.Builder newNotificationBuilder(Context context, Intent intent) {
+        final NotificationCompat.Builder builder = new NotificationCompat.Builder(context)
+                .setSmallIcon(R.drawable.app_icon)
+                .setAutoCancel(true)
+                .setDefaults(Notification.DEFAULT_ALL);
+        {
```

Comment:
I was trying to demarcate the code that i pulled straight from Parse's library source from our own code.